dipy/reconst/ivim.py

```diff
@@ -0,0 +1,391 @@
+
+
+class IvimModel(ReconstModel):
+    """Ivim model
+    """
+
+    def __init__(self, gtab, split_b=200.0,
+                 bounds=None, tol=1e-7,
+                 options={'gtol': 1e-7, 'ftol': 1e-7,
+                          'eps': 1e-7, 'maxiter': 1000}):
+        """
+        Initialize an IVIM model.
+
+        The IVIM model assumes that biological tissue includes a volume
+        fraction 'f' of water flowing in perfused capillaries, with a
+        perfusion coefficient D* and a fraction (1-f) of static (diffusion
+        only), intra and extracellular water, with a diffusion coefficient
+        D. In this model the echo attenuation of a signal in a single voxel
+        can be written as
+
+            .. math::
+
+            S(b) = S_0[f*e^{(-b*D\*)} + (1-f)e^{(-b*D)}]
+
+            Where:
+            .. math::
+
+            S_0, f, D\* and D are the IVIM parameters.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+            Gradient directions and bvalues
+
+        split_b : float
+            The b-value to split the data on for two-stage fit
+
+        bounds : array, (4,4)
+            Bounds for the parameters. This is applicable only for Scipy
+            version > 0.17 as we use least_squares for the fitting which
+            supports bounds. For versions less than Scipy 0.17, this is
+            by default set to None. Setting a bound on a Scipy version less
+            than 0.17 will raise an error.
+
+            This can be supplied as a tuple for Scipy versions 0.17. It is
+            recommended to upgrade to Scipy 0.17 for bounded fitting. The
+            default bounds are set to ([0., 0., 0., 0.], [np.inf, 1., 1., 1.])
+
+        tol : float, optional
+            Default : 1e-7
+            Tolerance for convergence of minimization.
+
+        options : dict, optional
+            Dictionary containing gtol, ftol, eps and maxiter. This is passed
+            to leastsq. By default these values are set to
+            options={'gtol': 1e-7, 'ftol': 1e-7, 'eps': 1e-7, 'maxiter': 1000}
+
+        References
+        ----------
+        .. [1] Le Bihan, Denis, et al. "Separation of diffusion
+                   and perfusion in intravoxel incoherent motion MR
+                   imaging." Radiology 168.2 (1988): 497-505.
+        .. [2] Federau, Christian, et al. "Quantitative measurement
+                   of brain perfusion with intravoxel incoherent motion
+                   MR imaging." Radiology 265.3 (2012): 874-881.
+        """
+        ReconstModel.__init__(self, gtab)
+        self.split_b = split_b
+        self.bounds = bounds
+        self.tol = tol
+        self.options = options
+
+        if SCIPY_LESS_0_17 and self.bounds is not None:
+            e_s = "Scipy versions less than 0.17 do not support "
+            e_s += "bounds. Please update to Scipy 0.17 to use bounds"
+            raise ValueError(e_s)
+        else:
```

Comment:
It looks like this resets the bounds or throws the scipy version error. If the user inputs bounds it seems like they get overwritten which isn't what we want. I think this should be elif self.bounds is None or maybe split the bounds check and scipy check into nested if statements.